types.cc

```diff
@@ -713,12 +713,10 @@ bool abstract_type::is_string() const {
     return visit(*this, visitor{});
 }
 
-static bool find(const abstract_type& t, std::function<bool(const abstract_type&)> f) {
-    if (f(t)) {
-        return true;
-    }
+template<typename T>
+static bool find(const abstract_type& t, const T& f) {
```

Comment:
Please add a concept for T (and a better name) so people know what to expect.

Reply:
I don't understand why this is a good thing. It bloats the runtime and compile time, to provide a performance improvement where one isn't needed.